hypershift-operator/controllers/nodepool/conditions.go

```diff
@@ -68,3 +78,38 @@ func findStatusCondition(conditions []hyperv1.NodePoolCondition, conditionType s
 
 	return nil
 }
+
+// generateReconciliationActiveCondition will generate the resource condition that reflects the state of reconciliation
+// on the resource.
+// (copied from support/util/pausereconcile_test.go and adjusted to use NodePoolCondition)
+func generateReconciliationActiveCondition(pausedUntilField *string, objectGeneration int64) hyperv1.NodePoolCondition {
```

Comment:
I know this is a copy of the hc func and makes sense to me to keep it consistent but as a note for the future, I find it unusual to not check the error right away, future TODO: refactor:
```
	isPaused, _, err := util.ProcessPausedUntilField(pausedUntilField, time.Now())
	if err != nil {
		return hyperv1.NodePoolCondition{
			Type:               hyperv1.NodePoolReconciliationActiveConditionType,
			Status:             corev1.ConditionTrue,
			Reason:             reconciliationInvalidPausedUntilConditionReason,
			Message:            "Invalid value provided for PausedUntil field",
			ObservedGeneration: objectGeneration,
		}
	}

	if isPaused {
		var msgString string
		if _, err := strconv.ParseBool(*pausedUntilField); err == nil {
			msgString = "Reconciliation paused until field removed"
		} else {
			msgString = fmt.Sprintf("Reconciliation paused until: %s", *pausedUntilField)
		}
		return hyperv1.NodePoolCondition{
			Type:               string(hyperv1.NodePoolReconciliationActiveConditionType),
			Status:             corev1.ConditionFalse,
			Reason:             reconciliationPausedConditionReason,
			Message:            msgString,
			ObservedGeneration: objectGeneration,
		}
	}

	return hyperv1.NodePoolCondition{
		Type:               hyperv1.NodePoolReconciliationActiveConditionType,
		Status:             corev1.ConditionTrue,
		Reason:             reconciliationActiveConditionReason,
		Message:            "Reconciliation active on resource",
		ObservedGeneration: objectGeneration,
	}
```

Reply:
Ack, I can look at pushing a follow-up PR which reworks this and the hc func :+1: